libs/webgui/pilight.js

```diff
@@ -353,6 +361,102 @@ function createPendingSwitchElement(sTabId, sDevId, aValues) {
 	}
 }
 
+function createContactElement(sTabId, sDevId, aValues) {
+	if($('#'+sDevId+'_contact').length == 0) {
+		if(bShowTabs) {
+			oTab = $('#'+sTabId).find('ul');
+		} else {
+			oTab = $('#all');
+		}
+		if('name' in aValues) {
+			oTab.append($('<li  id="'+sDevId+'" class="contact" data-icon="false"><div class="name">'+aValues['name']+'</div><div id="'+sDevId+'_contact" class="contact" data-role="fieldcontain" data-type="horizontal"><fieldset data-role="controlgroup" class="controlgroup" data-type="horizontal" data-mini="true"><input type="radio" name="'+sDevId+'_contact" id="'+sDevId+'_contact_closed" value="closed" /><label for="'+sDevId+'_contact_closed">'+language.closed+'</label><input type="radio" name="'+sDevId+'_contact" id="'+sDevId+'_contact_opened" value="opened" /><label for="'+sDevId+'_contact_opened">'+language.opened+'</label></fieldset></div></li>'));
+		}
+		$("div").trigger("create");
+		$('#'+sDevId+'_contact_closed').checkboxradio();
+		$('#'+sDevId+'_contact_opened').checkboxradio();
+		$('#'+sDevId+'_contact_closed').bind("change", function(event, ui) {
+			event.stopPropagation();
+			if('confirm' in aValues && aValues['confirm']) {
+				if(window.confirm("Are you sure?") == false) {
+					return false;
+				}
+			}
+			i = 0;
+			oLabel = this.parentNode.getElementsByTagName('label')[0];
+			$(oLabel).removeClass('ui-btn-active');
+			x = window.setInterval(function() {
+				i++;
+				if(i%2 == 1)
+					$(oLabel).removeClass('ui-btn-active');
+				else
+					$(oLabel).addClass('ui-btn-active');
+				if(i==2)
+					window.clearInterval(x);
+			}, 100);
+
+			if(oWebsocket) {
+				if('all' in aValues && aValues['all'] == 1) {
+					var json = '{"action":"control","code":{"device":"'+sDevId+'","state":"'+this.value+'","values":{"all": 1}}}';
+				} else {
+					var json = '{"action":"control","code":{"device":"'+sDevId+'","state":"'+this.value+'"}}'
+				}
+				oWebsocket.send(json);
+			} else {
+				bSending = true;
+				if('all' in aValues && aValues['all'] == 1) {
+					$.get(sHTTPProtocol+'://'+location.host+'/control/control?device='+sDevId+'&state='+this.value+'&values[all]=1');
```

Comment:
The URL is wrong, double `/control`.
Just saw that this is wrong in `createScreenElement()` too, on line 399. But that's unrelated to this pr.